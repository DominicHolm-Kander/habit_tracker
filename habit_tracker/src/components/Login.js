import React, { Component } from 'react'

export class Login extends Component {

    handleChangeOfPassword = (event) => {
        this.setState({currenInputPassword: event.target.value})
    }
    handleChangeOfUsername = (event) => {
        this.setState({currenInputUsername: event.target.value})
    }


    render() {


        return (
            <div>
                <h3>Put your login info here</h3>
                <label for="userName">Username
                <input type="text" className = "userName" name="userName" onChange={(event) => this.handleChangeOfUsername(event)}></input>
                </label>
                <label for="password">Password
                <input type="text" className = "password" name="password" onChange={(event) => this.handleChangeOfPassword(event)}></input>
                </label>
                <button type="button" id="loginButton" onClick={() => this.props.checkCredentials(this.state.currenInputUsername, this.state.currenInputPassword)}>Log in</button>
            </div
        )
    }
}

export default Login
