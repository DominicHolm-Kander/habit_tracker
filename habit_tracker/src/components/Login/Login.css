.login {
    display: flex;
    flex-direction: row;
}

.loginButton {
    outline: none;
    height: 50px;
    width: 100px;
    box-sizing: border-box;
    appearance: none;
    background-color: transparent;
    border-radius: 0.6em;
    cursor: pointer;
    display: flex;
    align-self: center;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1;
    margin: 5px;
    padding: 1.2em 1.2em;
    text-decoration: none;
    text-align: center;
    text-transform: uppercase;
    font-family: 'Montserrat', sans-serif;
    font-weight: 700;
    border-color: rgb(0, 255, 64);
    color: #fff;
    box-shadow: 0 0 40px 40px rgb(39, 182, 26) inset, 0 0 0 0 rgb(20, 214, 69);
    transition: all 150ms ease-in-out;
}
  .loginButton:hover {
    box-shadow: 0 0 10px 0 rgb(47, 201, 33) inset, 0 0 10px 4px rgb(30, 189, 91);
    color: green;
}

