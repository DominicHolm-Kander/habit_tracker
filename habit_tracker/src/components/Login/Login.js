import React, { Component } from 'react'
import './Login.css'
// import { FormControl } from '@material-ui/core';

export class Login extends Component {

    handleChangeOfPassword = (event) => {
        this.setState({currenInputPassword: event.target.value})
    }
    handleChangeOfUsername = (event) => {
        this.setState({currenInputUsername: event.target.value})
    }


    render() {


        return (
            <div className="login">
            <h3>Sign in</h3>
            <div className="container">
        <div className="form-group">
            <label for="exampleInputEmail1">Username</label>
            <input type="text" className="form-control" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Username" onChange={(event) => this.handleChangeOfUsername(event)}></input>
        </div>
        <div className="form-group">
            <label for="exampleInputPassword1">Password</label>
            <input type="password" className="form-control" id="exampleInputPassword1" placeholder="Password" onChange={(event) => this.handleChangeOfPassword(event)}></input>
            <small id="emailHelp" className="form-text text-muted">We'll never share your password with anyone else.</small>
        </div>
        </div>
        <button type="submit" className="btn btn-primary" onClick={() => this.props.checkCredentials(this.state.currenInputUsername, this.state.currenInputPassword)}>Submit</button>
       
            </div>
        )
            
        }    
    
    }

export default Login
