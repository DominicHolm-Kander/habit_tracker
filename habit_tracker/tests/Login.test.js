import React from 'react'
import { shallow } from 'enzyme'
import Login from "../src/components/Login/Login"
import waitUntil from "async-wait-until"; // For async testing of setState functions.
import regeneratorRuntime from 'regenerator-runtime';

describe("shallow Login", () => {
    let wrapper;
    let checkCredentialsMock = jest.fn()

    beforeEach(() => (wrapper = shallow(<Login checkCredentials={checkCredentialsMock}/>)))
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("should have 2 input fields and a button", () => {
        expect(wrapper.find("input").length).toEqual(2);
        expect(wrapper.find("button").length).toEqual(1);
    })
    it("loginButton should call 'checkCredentials' function when clicked", 
    async () => {
        wrapper.setState({currentInputUsername: "placeholder", currentInputPassword: "alsoAPlaceholder"})
        // Adding placeholder data is needed to stop the function throwing a TypeError when given parameters equal to null
        await waitUntil(() => wrapper.instance().state.currentInputPassword && wrapper.instance().state.currentInputUsername);        expect(checkCredentialsMock).not.toHaveBeenCalled();
        wrapper.find('button').simulate('click');
        expect(checkCredentialsMock).toHaveBeenCalled();
    })
    it("HandleChangeOfPassword function should update component state", async () => {
        wrapper.instance().handleChangeOfPassword('testPassword');
        await waitUntil(() => wrapper.instance().state.currentInputPassword);
        expect(wrapper.instance().state.currentInputPassword).toEqual('testPassword');
    })
    it("HandleChangeOfUsername function should update component state", async () => {
        wrapper.instance().handleChangeOfUsername('testUsernameValue');
        await waitUntil(() => wrapper.instance().state.currentInputUsername);
        expect(wrapper.instance().state.currentInputUsername).toEqual('testUsernameValue');
    })
    it("Typing in the username input element should call handleChangeOfusername function", 
    () => {
        let spy = jest.spyOn(wrapper.instance(), "handleChangeOfUsername");
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.find('[name="userName"]').at(0)
        .simulate("change", { target: { value: "anything"}});
        expect(spy).toHaveBeenCalledTimes(1)
    })
    it("Typing in a password should call handleChangeOfPassword function", 
    () => {
        let spy = jest.spyOn(wrapper.instance(), "handleChangeOfPassword");
        wrapper.instance().forceUpdate();  
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.find('[name="password"]').at(0)
        .simulate("change", { target: { value: "whateverYouPlease"}});
        expect(spy).toHaveBeenCalledTimes(1)
    })
})