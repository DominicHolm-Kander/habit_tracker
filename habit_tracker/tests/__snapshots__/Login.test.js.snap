// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Login should render correctly 1`] = `
<div
  className="login"
>
  <h3>
    Sign in
  </h3>
  <div
    className="container"
  >
    <div
      className="form-group"
    >
      <label
        for="exampleInputEmail1"
      >
        Username
      </label>
      <input
        aria-describedby="emailHelp"
        className="form-control"
        id="exampleInputEmail1"
        name="userName"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        for="exampleInputPassword1"
      >
        Password
      </label>
      <input
        className="form-control"
        id="exampleInputPassword1"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <small
        className="form-text text-muted"
        id="emailHelp"
      >
        We'll never share your password with anyone else.
      </small>
    </div>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
</div>
`;
